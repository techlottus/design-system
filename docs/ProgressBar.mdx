---
name: ProgressBar
menu: Components
---

import { Playground, Props } from 'docz'
import ProgressBar from '../packages/ProgressBar'

# ProgressBar

ProgressBar component docz ðŸ“„

## Basic Usage

<Playground>
  <ProgressBar />
</Playground>