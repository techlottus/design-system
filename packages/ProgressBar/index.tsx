import * as React from 'react';
import classNames from 'classnames';
import { variantIntensity, variantName } from '../common/constants';

interface IProgressBarProps {
  value: number;
  variant?: string;
  variantAs?: string;
}

const ProgressBar: React.SFC<IProgressBarProps> = ({value, variant, variantAs}: IProgressBarProps) => {
  const classes = classNames(
    'h-full rounded',
    `bg-${variantName[variant]}-${variantIntensity[variantAs]}`,
  );
  return (
    <div className='w-full h-2 bg-surface-100 rounded'>
      <div
        className={classes}
        style={{width: `${value}%`, maxWidth: '100%'}}>
      </div>
    </div>
  );
};

ProgressBar.defaultProps = {
  value: 0,
  variant: 'primary',
  variantAs: '200',
};

export default ProgressBar;
