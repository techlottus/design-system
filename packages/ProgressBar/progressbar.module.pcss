.progress {
  @apply w-full h-2 bg-surface-100 rounded;
  &Bar {
    @apply h-full rounded;
  }
}