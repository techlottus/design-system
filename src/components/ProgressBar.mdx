---
name: ProgressBar
menu: Components
---

import { Props } from 'docz'
import ProgressBar from '../packages/ProgressBar';
import ProgressBarExample from './examples/ProgressBarExamples.tsx'


# ProgressBar

ProgressBar component docs ðŸ“„

## Props
<Props of={ProgressBar} />

## Basic Usage
```jsx
<ProgressBar value={30} variant='technology' variantAs='300'/>
```
<ProgressBar value={30} variant='technology' variantAs='300'/>

## Example
<ProgressBarExample />
