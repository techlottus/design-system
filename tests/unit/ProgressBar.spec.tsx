import * as React from 'react';
import { shallow, mount } from 'enzyme';
import ProgressBar from '../../packages/ProgressBar';

describe('ProgressBar.tsx', () => {
  it('Should match the snapshot', () => {
    const vm = shallow(<ProgressBar value={25} />);
    expect(vm).toMatchSnapshot();
  });

  it('the component ProgressBar has the correct className', () => {
    const vm = shallow(<ProgressBar value={30} />);
    expect(vm.props().className).toBe('w-full h-2 bg-surface-100 rounded');
  });

  it('when the component is created without a "variant" prop, the default variant is "primary"', () => {
    const vm = mount(<ProgressBar value={50} />);
    expect(vm.props().variant).toBe('primary');
  });

  it('when the component is created without a "variantAs" prop, the default variantAs is "200"', () => {
    const vm = mount(<ProgressBar value={50} /> );
    expect(vm.props().variantAs).toBe('200');
  });
});
