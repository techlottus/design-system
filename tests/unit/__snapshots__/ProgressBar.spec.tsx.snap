// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar.tsx Should match the snapshot 1`] = `
<div
  className="w-full h-2 bg-surface-100 rounded"
>
  <div
    className="h-full rounded bg-primary-200"
    style={
      Object {
        "width": "25%",
      }
    }
  />
</div>
`;
